ext/standard/array.c

```diff
@@ -2555,7 +2555,7 @@ static void php_compact_var(HashTable *eg_active_symbol_table, zval *return_valu
 	} else if (Z_TYPE_P(entry) == IS_ARRAY) {
 	    if (Z_REFCOUNTED_P(entry)) {
 			if (Z_IS_RECURSIVE_P(entry)) {
-				zend_throw_error(NULL, "Recursion detected");
+				zend_value_error("Recursion detected");
```

Comment:
I suppose the recursion case should also be a normal error instead of a value error?